opensearch-operator/pkg/reconcilers/dashboards_test.go

```diff
@@ -298,6 +298,118 @@ var _ = Describe("Dashboards Reconciler", func() {
+						Labels: map[string]string{
+							"testLabelKey":  "testValue",
+							"testLabelKey2": "testValue2",
+						},
+					},
+				}}
+			ns := corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterName,
+				},
+			}
+			err := k8sClient.Create(context.Background(), &ns)
+			Expect(err).ToNot(HaveOccurred())
+
+			reconcilerContext := NewReconcilerContext(spec.Spec.NodePools)
+			underTest := NewDashboardsReconciler(
+				k8sClient,
+				context.Background(),
+				&helpers.MockEventRecorder{},
+				&reconcilerContext,
+				&spec,
+			)
+			_, err = underTest.Reconcile()
+			Expect(err).ToNot(HaveOccurred())
+			deployment := appsv1.Deployment{}
+			Eventually(func() bool {
+				err := k8sClient.Get(context.Background(), client.ObjectKey{Name: clusterName + "-dashboards", Namespace: clusterName}, &deployment)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+			Eventually(Object(&appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      clusterName + "-dashboards",
+					Namespace: clusterName,
+				},
+			}, k8sClient), timeout, interval).Should(ExistAnd(
+				HaveMatchingContainer(
+					HaveLabels(
+						"testLabelKey",
+						"testValue",
+						"testLabelKey2",
+						"testValue2",
+					),
+				),
+			))
+		})
+	})
+
+	When("running the dashboards reconciler with annotations supplied", func() {
+		It("should populate the dashboard pod spec with annotations", func() {
+			clusterName := "dashboards-add-annotations"
+			spec := opsterv1.OpenSearchCluster{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName, Namespace: clusterName, UID: "dummyuid"},
+				Spec: opsterv1.ClusterSpec{
+					General: opsterv1.GeneralConfig{ServiceName: clusterName},
+					Dashboards: opsterv1.DashboardsConfig{
+						Enable: true,
+						Labels: map[string]string{
+							"testAnnotationKey":  "testValue",
+							"testAnnotationKey2": "testValue2",
+						},
+					},
+				}}
+			ns := corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterName,
+				},
+			}
+			err := k8sClient.Create(context.Background(), &ns)
+			Expect(err).ToNot(HaveOccurred())
+
+			reconcilerContext := NewReconcilerContext(spec.Spec.NodePools)
+			underTest := NewDashboardsReconciler(
+				k8sClient,
+				context.Background(),
+				&helpers.MockEventRecorder{},
+				&reconcilerContext,
+				&spec,
+			)
+			_, err = underTest.Reconcile()
+			Expect(err).ToNot(HaveOccurred())
+			deployment := appsv1.Deployment{}
+			Eventually(func() bool {
+				err := k8sClient.Get(context.Background(), client.ObjectKey{Name: clusterName + "-dashboards", Namespace: clusterName}, &deployment)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+			Eventually(Object(&appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      clusterName + "-dashboards",
+					Namespace: clusterName,
+				},
+			}, k8sClient), timeout, interval).Should(ExistAnd(
+				HaveMatchingContainer(
+					HaveLabels(
```

Comment:
Shouldn't this check annotations instead of labels?

Reply:
Thank you for pointing that out!  Ate too much turkey during Thanksgiving and missed this!  After looking at the available matchers in kmatch I will have to implement the test for annotations in the builders package in a new test file for dashboards.

I should have an update later today, and I will also update the userguide as requested.  Thanks for the feedback!